test/Dialect/FIRRTL/inliner.mlir

```diff
@@ -753,3 +753,36 @@ firrtl.circuit "Issue3334_flatten" {
     firrtl.instance foo sym @foo @Foo()
   }
 }
+
+firrtl.circuit "instNameRename"  {
+  firrtl.hierpath @nla_5560 [@instNameRename::@bar0, @Bar0::@w, @Bar2::@w, @Bar1::@a]
+  // CHECK:  firrtl.hierpath @nla_5560 [@instNameRename::@[[w_1:.+]], @Bar2::@w, @Bar1::@a]
+  firrtl.hierpath @nla_5560_1 [@instNameRename::@bar1, @Bar0::@w, @Bar2::@w, @Bar1::@a]
+  // CHECK:  firrtl.hierpath @nla_5560_1 [@instNameRename::@[[w_2:.+]], @Bar2::@w, @Bar1::@a]
+
+  firrtl.module @Leaf() {
+    %w = firrtl.wire   : !firrtl.uint<8>
+    %inv = firrtl.invalidvalue : !firrtl.uint<8>
+    firrtl.strictconnect %w, %inv : !firrtl.uint<8>
+  }
+  firrtl.module @Bar1() {
+    firrtl.instance leaf sym @a  {annotations = [{circt.nonlocal = @nla_5560, class = "test0"},{circt.nonlocal = @nla_5560_1, class = "test1"}]} @Leaf()
```

Comment:
This crash exists _without_ the annotations, so no need to continue to preserve old-style NLAs in new tests:

```suggestion
    firrtl.instance leaf sym @a @Leaf()
```